src/parser.rs

```diff
@@ -707,6 +735,12 @@ impl<'a> Parser<'a> {
             }
         } else if Token::DoubleColon == tok {
             self.parse_pg_cast(expr)
+        } else if Token::ExclamationMark == tok {
+            // PostgreSQL factorial operation
+            Ok(Expr::UnaryOp {
```

Comment:
Ideally we'd consider how postfix factorial interacted with other operators (the postfix `::type`, another factorial, and prefix operators), but I don't think this has to block merging this PR.